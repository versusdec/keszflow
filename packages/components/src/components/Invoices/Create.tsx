import React, { useState } from 'react'
import {
  Box,
  Dialog,
  DialogTitle,
  DialogContent,
  IconButton,
  Stack,
  Button,
  Divider,
  DialogActions,
  Typography,
  TextField,
  MenuItem,
  Select,
  FormControl,
  InputLabel,
} from '@mui/material'
import { Close } from '@mui/icons-material'
import * as Yup from 'yup'
import moment from 'moment'
import { DesktopDatePicker, DatePicker } from '@mui/x-date-pickers'
import {
  Formik,
  Form,
  FormikHelpers,
  FormikProps,
  FieldArray,
  ArrayHelpers,
} from 'formik'

import { Input } from '../../elements/input'
import { ActionButton } from '../../elements/actionButton'
import Sortable from '../Sortable'
import { useInvoice } from '../../../../panel/hooks/useInvoice'

interface CreateModalProps {
  open: boolean
  onClose: () => void
  item?: Invoice
  id?: number
}

export const useCreateModal = () => {
  const [open, setOpen] = useState(false)

  const handleCreateModal = () => {
    setOpen(!open)
  }

  return { open, handleCreateModal }
}

interface DatesInfo {
  issue: string | null
  end: string | null
  due: string | null
}

export interface ItemInfo {
  pos: number
  name: string
  unit: string
  quantity: number | ''
  netPrice: number | ''
  taxRate: number | ''
  netAmount: number | ''
  taxAmount: number | ''
  grossAmount: number | ''
}

export interface SellerInfo {
  name: string
  country: 'Poland' | 'Ukraine'
  address_line_1: string
  address_line_2?: string
  address_line_3?: string
}

export interface Invoice {
  no: string
  seller: SellerInfo
  buyer: SellerInfo
  dates: DatesInfo
  payment: 'cash' | 'bank' | string
  items: ItemInfo[]
}

export const InvoiceCreate = ({ open, onClose, id }: CreateModalProps) => {
  const res = id && useInvoice(id)

  const item = res && res.data
  const isFetching = res && res.isFetching
  const isError = res && res.isError

  const defaultItem: ItemInfo = {
    pos: 0,
    name: '',
    unit: '',
    quantity: '',
    netPrice: '',
    taxRate: '',
    netAmount: '',
    taxAmount: '',
    grossAmount: '',
  }
  const invoiceValues: Invoice = item || {
    no: '',
    seller: {
      name: '',
      country: 'Poland',
      address_line_1: '',
      address_line_2: '',
      address_line_3: '',
    },
    buyer: {
      name: '',
      country: 'Poland',
      address_line_1: '',
      address_line_2: '',
      address_line_3: '',
    },
    dates: {
      issue: null,
      end: null,
      due: null,
    },
    payment: 'cash',
    items: [defaultItem],
  }

  const validationSchema = Yup.object({
    dates: Yup.object({
      issue: Yup.date().nullable(),
    }),
    client: Yup.string().required('This field is required'),
    seller: Yup.string().required('This field is required'),
    create_date: Yup.string().required('This field is required'),
    payment_date: Yup.string().required('This field is required'),
    payment_type: Yup.string()
      .oneOf(['bank', 'cash'])
      .required('This field is required'),
    items: Yup.array().of(
      Yup.object({
        name: Yup.string().required('This field is required'),
        amount: Yup.number().required('This field is required'),
        price: Yup.number().required('This field is required'),
        netto: Yup.number().required('This field is required'),
        vat: Yup.number().required('This field is required'),
        quota: Yup.number().required('This field is required'),
        brutto: Yup.number().required('This field is required'),
      })
    ),
  })

  const itemJSX = ({ item, index, formikProps }: any) => {
    return (
      <Stack direction={'row'} spacing={2} mb={2} key={index}>
        <Box>
          <ActionButton onClick={() => {}} icon="apps" tooltip={'Drag'} />
        </Box>
        <Box>
          <Input
            label={'Name'}
            fast="true"
            name={`items.${index}.name`}
            value={`items.${index}.name` || ''}
            size={'small'}
          />
        </Box>
        <Box>
          <Input
            label={'Unit'}
            fast="true"
            name={`items.${index}.unit`}
            value={`items.${index}.unit`}
            size={'small'}
          />
        </Box>
        <Box>
          <Input
            label={'Quantity'}
            fast="true"
            name={`items.${index}.quantity`}
            value={`items.${index}.quantity`}
            size={'small'}
          />
        </Box>

        <Box>
          <Input
            label={'Net price'}
            fast="true"
            name={`items.${index}.netPrice`}
            value={`items.${index}.netPrice`}
            size={'small'}
          />
        </Box>
        <Box>
          <Input
            label={'Net amount'}
            fast="true"
            name={`items.${index}.netAmount`}
            value={`items.${index}.netAmount`}
            size={'small'}
          />
        </Box>
        <Box>
          <Input
            label={'Tax rate'}
            fast="true"
            name={`items.${index}.taxRate`}
            value={`items.${index}.taxRate`}
            size={'small'}
          />
        </Box>
        <Box>
          <Input
            label={'Tax amount'}
            fast="true"
            name={`items.${index}.taxAmount`}
            value={`items.${index}.taxAmount`}
            size={'small'}
          />
        </Box>
        <Box>
          <Input
            label={'Gross amount'}
            fast="true"
            name={`items.${index}.grossAmount`}
            value={`items.${index}.grossAmount`}
            size={'small'}
          />
        </Box>
        <Box justifyContent={'flex-end'}>
          <ActionButton
            onClick={() => {
              const newItems = [...formikProps.values.items]
              newItems.splice(index, 1)
              formikProps.setFieldValue('items', newItems)
            }}
            disabled={formikProps.values.items.length === 1}
            icon="delete"
            tooltip={'Remove'}
          />
        </Box>
      </Stack>
    )
  }
  const ModalJSX = (
    <>
      <Dialog
        open={open}
        onClose={onClose}
        fullWidth={true}
        scroll={'paper'}
        maxWidth={'lg'}
      >
        <Formik
          initialValues={invoiceValues}
          // validationSchema={validationSchema}
          // enableReinitialize={true}
          onSubmit={(
            values: Invoice,
            formikHelpers: FormikHelpers<Invoice>
          ) => {
            values.items &&
              values.items.map((item, i) => {
                item.pos = i
              })
            console.log(values)
            formikHelpers.setSubmitting(false)
          }}
        >
          {(formikProps: FormikProps<Invoice>) => (
            <Form>
              <DialogTitle sx={{ pr: 10 }}>
                <IconButton
                  aria-label="close"
                  onClick={onClose}
                  sx={{
                    position: 'absolute',
                    right: 8,
                    top: 8,
                    color: (theme) => theme.palette.primary.main,
                  }}
                >
                  <Close />
                </IconButton>
                Create Invoice
              </DialogTitle>
              <DialogContent dividers>
                <Box sx={{}}>
                  <Stack spacing={2} direction={'row'}>
                    <Input label={'Number'} required name={'no'} />
                  </Stack>
                  <Box sx={{ mt: 2, mb: 3 }}>
                    <Typography variant={'h6'}>Seller</Typography>
                    <Divider />
                  </Box>
                  <Stack direction="row" spacing={2}>
                    <Input label={'Name'} required name={'seller.name'} />
                    <Input
                      label={'Address 1'}
                      required
                      name={'seller.address_line_1'}
                    />
                    <Input label={'Address 2'} name={'seller.address_line_2'} />
                    <Input label={'Address 3'} name={'seller.address_line_3'} />
                    <FormControl>
                      <InputLabel id="seller.country">Country</InputLabel>
                      <Select
                        id={'seller.country'}
                        name={'seller.country'}
                        value={formikProps.values.seller.country}
                        label={'country'}
                        onChange={(e) => {
                          formikProps.setFieldValue(
                            'seller.country',
                            e.target.value
                          )
                        }}
                      >
                        <MenuItem value={'Poland'}>Poland</MenuItem>
                        <MenuItem value={'Ukraine'}>Ukraine</MenuItem>
                      </Select>
                    </FormControl>
                  </Stack>
                  <Box sx={{ mt: 2, mb: 3 }}>
                    <Typography variant={'h6'}>Buyer</Typography>
                    <Divider />
                  </Box>
                  <Stack direction="row" spacing={2}>
                    <Input label={'Name'} required name={'buyer.name'} />
                    <Input
                      label={'Address 1'}
                      required
                      name={'buyer.address_line_1'}
                    />
                    <Input label={'Address 2'} name={'buyer.address_line_2'} />
                    <Input label={'Address 3'} name={'buyer.address_line_3'} />
                    <FormControl>
                      <InputLabel id="buyer.country">Country</InputLabel>
                      <Select
                        id={'buyer.country'}
                        name={'buyer.country'}
                        value={formikProps.values.buyer.country}
                        label={'country'}
                        onChange={(e) => {
                          formikProps.setFieldValue(
                            'buyer.country',
                            e.target.value
                          )
                        }}
                      >
                        <MenuItem value={'Poland'}>Poland</MenuItem>
                        <MenuItem value={'Ukraine'}>Ukraine</MenuItem>
                      </Select>
                    </FormControl>
                  </Stack>
                  <Divider sx={{ mt: 2, mb: 3 }} />
                  <Stack direction="row" spacing={2}>
                    <FormControl>
                      <InputLabel id="payment">Payment method</InputLabel>
                      <Select
                        id={'payment'}
                        name={'payment'}
                        value={formikProps.values.payment}
                        label={'Payment method'}
                        onChange={(e) => {
                          formikProps.setFieldValue('payment', e.target.value)
                        }}
                      >
                        <MenuItem value={'cash'}>Cash</MenuItem>
                        <MenuItem value={'bank'}>Bank</MenuItem>
                      </Select>
                    </FormControl>
                    <DatePicker
                      label="Issue date"
                      // inputFormat="DD.MM.yyyy"
                      value={formikProps.values.dates.issue}
                      onChange={(val) => {
                        formikProps.setFieldValue(
                          'dates.issue',
                          moment(val).toISOString()
                        )
                      }}
                      renderInput={(params) => <TextField {...params} />}
                    />
                    <DatePicker
                      label="End date"
                      // inputFormat="DD.MM.yyyy"
                      value={formikProps.values.dates.end}
                      onChange={(val) => {
                        formikProps.setFieldValue(
                          'dates.end',
                          moment(val).toISOString()
                        )
                      }}
                      renderInput={(params) => <TextField {...params} />}
                    />
                    <DatePicker
                      label="Due date"
                      // inputFormat="DD.MM.yyyy"
                      value={formikProps.values.dates.due}
                      onChange={(val) => {
                        formikProps.setFieldValue(
                          'dates.due',
                          moment(val).toISOString()
                        )
                      }}
                      renderInput={(params) => <TextField {...params} />}
                    />
                  </Stack>
                  <Divider sx={{ mt: 2, mb: 3 }} />
                  <Stack
                    direction="row"
                    justifyContent={'space-between'}
                    spacing={2}
                  >
                    <Typography variant={'h6'}>Items</Typography>
                    <Button
                      type={'button'}
                      variant={'contained'}
                      sx={{}}
                      onClick={() => {
                        const newItems = [...formikProps.values.items]
                        newItems.push({
                          pos: formikProps.values.items.length,
                          name: '',
                          unit: '',
                          quantity: '',
                          netPrice: '',
                          taxRate: '',
                          netAmount: '',
                          taxAmount: '',
                          grossAmount: '',
                        })
                        formikProps.setFieldValue('items', newItems)
                      }}
                    >
                      Add
                    </Button>
                  </Stack>
                  <Divider sx={{ mt: 2, mb: 3 }} />
                  <Sortable
                    items={formikProps.values.items}
                    Component={itemJSX}
                    formikProps={formikProps}
                    onChange={(items: any[]) => {
                      formikProps.setFieldValue('items', items)
                    }}
                  />
                </Box>
              </DialogContent>
              <DialogActions>
                <Button type={'submit'} variant={'contained'} sx={{}}>
                  Save
                </Button>
              </DialogActions>
            </Form>
          )}
        </Formik>
      </Dialog>
    </>
  )
  return <>{!isFetching && ModalJSX}</>
}
