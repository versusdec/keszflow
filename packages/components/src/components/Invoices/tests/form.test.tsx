import React from 'react'
import { InvoiceForm } from '../Form'
import {
  fireEvent,
  render,
  cleanup,
  waitFor,
  act,
} from '@testing-library/react'
import { mount, shallow, ReactWrapper } from 'enzyme'
import { IInvoice, IItemInfo } from '../Create'
import { LocalizationProvider } from '@mui/x-date-pickers'
import { AdapterMoment } from '@mui/x-date-pickers/AdapterMoment'

afterEach(cleanup)

const defaultItem: IItemInfo = {
  pos: 0,
  name: '',
  unit: '',
  quantity: '',
  netPrice: '',
  taxRate: '',
  netAmount: '',
  taxAmount: '',
  grossAmount: '',
}
const fakeValues = {
  no: '100500',
  seller: {
    name: 'Konyaka',
    country: 'Poland',
    address_line_1: 's1',
    address_line_2: 's2',
    address_line_3: 's3',
  },
  buyer: {
    name: 'skotynyaka',
    country: 'Poland',
    address_line_1: 'b1',
    address_line_2: 'b2',
    address_line_3: 'b3',
  },
  dates: {
    issue: '20/07/2022',
    end: '20/07/2022',
    due: '20/07/2022',
  },
  payment: 'cash',
}

const props = {
  invoice: undefined,
}

describe('Invoice Form render', () => {
  const getComponent = () => (
    <LocalizationProvider dateAdapter={AdapterMoment}>
      <InvoiceForm {...props} />
    </LocalizationProvider>
  )

  it('Should render correctly with Shallow', () => {
    const component = shallow(getComponent())
    expect(component).toMatchSnapshot()
  })

  it('Should render correctly with mount', () => {
    const component = mount(getComponent())
    expect(component).toMatchSnapshot()
  })
})

describe('render all inputs', () => {
  let no: HTMLInputElement | null,
    sellerName: HTMLInputElement | null,
    sellerAddress1: HTMLInputElement | null,
    sellerAddress2: HTMLInputElement | null,
    sellerAddress3: HTMLInputElement | null,
    sellerCountry: HTMLInputElement | null,
    buyerName: HTMLInputElement | null,
    buyerAddress1: HTMLInputElement | null,
    buyerAddress2: HTMLInputElement | null,
    buyerAddress3: HTMLInputElement | null,
    buyerCountry: HTMLInputElement | null,
    paymentMethod: HTMLInputElement | null,
    dueDate: HTMLInputElement | null,
    endDate: HTMLInputElement | null,
    issueDate: HTMLInputElement | null

  const getComponent = () => (
    <LocalizationProvider dateAdapter={AdapterMoment}>
      <InvoiceForm {...props} />
    </LocalizationProvider>
  )

  beforeEach(() => {
    const { container } = render(getComponent())
    no = container.querySelector('input[name=no]')
    sellerName = container.querySelector('input[name="seller.name"]')
    sellerAddress1 = container.querySelector(
      'input[name="seller.address_line_1"]'
    )
    sellerAddress2 = container.querySelector(
      'input[name="seller.address_line_2"]'
    )
    sellerAddress3 = container.querySelector(
      'input[name="seller.address_line_3"]'
    )
    sellerCountry = container.querySelector('input[name="seller.name"]')
    buyerName = container.querySelector('input[name="buyer.name"]')
    buyerAddress1 = container.querySelector(
      'input[name="buyer.address_line_1"]'
    )
    buyerAddress2 = container.querySelector(
      'input[name="buyer.address_line_2"]'
    )
    buyerAddress3 = container.querySelector(
      'input[name="buyer.address_line_3"]'
    )
    buyerCountry = container.querySelector('input[name="buyer.name"]')
    paymentMethod = container.querySelector('input[name="payment"]')
    dueDate = container.querySelector('.dates_due input')
    issueDate = container.querySelector('.dates_issue input')
    endDate = container.querySelector('.dates_end input')
  })

  it('renders all inputs', () => {
    expect(no).toBeInTheDocument()
    expect(sellerName).toBeInTheDocument()
    expect(sellerAddress1).toBeInTheDocument()
    expect(sellerAddress2).toBeInTheDocument()
    expect(sellerAddress3).toBeInTheDocument()
    expect(sellerCountry).toBeInTheDocument()
    expect(buyerName).toBeInTheDocument()
    expect(buyerAddress1).toBeInTheDocument()
    expect(buyerAddress2).toBeInTheDocument()
    expect(buyerAddress3).toBeInTheDocument()
    expect(buyerCountry).toBeInTheDocument()
    expect(paymentMethod).toBeInTheDocument()
    expect(dueDate).toBeInTheDocument()
    expect(issueDate).toBeInTheDocument()
    expect(endDate).toBeInTheDocument()

    // expect(no?.value).toMatch('')
  })
})

describe('input values', () => {
  let no: ReactWrapper,
    sellerName: ReactWrapper,
    sellerAddress1: ReactWrapper,
    sellerAddress2: ReactWrapper,
    sellerAddress3: ReactWrapper,
    sellerCountry: ReactWrapper,
    buyerName: ReactWrapper,
    buyerAddress1: ReactWrapper,
    buyerAddress2: ReactWrapper,
    buyerAddress3: ReactWrapper,
    buyerCountry: ReactWrapper,
    paymentMethod: ReactWrapper,
    dueDate: ReactWrapper,
    endDate: ReactWrapper,
    issueDate: ReactWrapper

  const getComponent = () => (
    <LocalizationProvider dateAdapter={AdapterMoment}>
      <InvoiceForm {...props} />
    </LocalizationProvider>
  )

  beforeEach(() => {
    const container = mount(getComponent())
    no = container.find('input[name="no"]')
    sellerName = container.find('input[name="seller.name"]')
    sellerAddress1 = container.find('input[name="seller.address_line_1"]')
    sellerAddress2 = container.find('input[name="seller.address_line_2"]')
    sellerAddress3 = container.find('input[name="seller.address_line_3"]')
    sellerCountry = container.find('input[name="seller.name"]')
    buyerName = container.find('input[name="buyer.name"]')
    buyerAddress1 = container.find('input[name="buyer.address_line_1"]')
    buyerAddress2 = container.find('input[name="buyer.address_line_2"]')
    buyerAddress3 = container.find('input[name="buyer.address_line_3"]')
    buyerCountry = container.find('input[name="buyer.name"]')
    paymentMethod = container.find('input[name="payment"]')
    dueDate = container.find('.dates_due input')
    issueDate = container.find('.dates_issue input')
    endDate = container.find('.dates_end input')
  })

  it('fill values', () => {
    no.simulate('change', {
      persist: () => {},
      target: {
        value: fakeValues.no,
      },
    })
    sellerName.simulate('change', {
      persist: () => {},
      target: {
        value: fakeValues.seller.name,
      },
    })
    sellerAddress1.simulate('change', {
      persist: () => {},
      target: {
        value: fakeValues.seller.address_line_1,
      },
    })
    sellerAddress2.simulate('change', {
      persist: () => {},
      target: {
        value: fakeValues.seller.address_line_2,
      },
    })
    sellerAddress3.simulate('change', {
      persist: () => {},
      target: {
        value: fakeValues.seller.address_line_3,
      },
    })
    sellerCountry.simulate('change', {
      persist: () => {},
      target: {
        value: fakeValues.seller.country,
      },
    })
    buyerName.simulate('change', {
      persist: () => {},
      target: {
        value: fakeValues.buyer.name,
      },
    })
    buyerAddress1.simulate('change', {
      persist: () => {},
      target: {
        value: fakeValues.buyer.address_line_1,
      },
    })
    buyerAddress2.simulate('change', {
      persist: () => {},
      target: {
        value: fakeValues.buyer.address_line_2,
      },
    })
    buyerAddress3.simulate('change', {
      persist: () => {},
      target: {
        value: fakeValues.buyer.address_line_3,
      },
    })
    buyerCountry.simulate('change', {
      persist: () => {},
      target: {
        value: fakeValues.buyer.country,
      },
    })
    paymentMethod.simulate('change', {
      persist: () => {},
      target: {
        value: fakeValues.payment,
      },
    })
    issueDate.simulate('change', {
      persist: () => {},
      target: {
        value: fakeValues.dates.issue,
      },
    })
    endDate.simulate('change', {
      persist: () => {},
      target: {
        value: fakeValues.dates.end,
      },
    })
    dueDate.simulate('change', {
      persist: () => {},
      target: {
        value: fakeValues.dates.due,
      },
    })
    expect(no.html()).toMatch(fakeValues.no)
  })
})

describe('FormComponent', () => {
  describe('Submitting form', () => {
    // Arrange--------------
    // Set up variables accessible in tests
    let container
    let no: HTMLInputElement,
      sellerName: HTMLInputElement,
      sellerAddress1: HTMLInputElement,
      sellerAddress2: HTMLInputElement,
      sellerAddress3: HTMLInputElement,
      sellerCountry: HTMLInputElement,
      buyerName: HTMLInputElement,
      buyerAddress1: HTMLInputElement,
      buyerAddress2: HTMLInputElement,
      buyerAddress3: HTMLInputElement,
      buyerCountry: HTMLInputElement,
      paymentMethod: HTMLInputElement,
      dueDate: HTMLInputElement,
      endDate: HTMLInputElement,
      issueDate: HTMLInputElement,
      form: HTMLFormElement,
      handleSubmit: () => void,
      fakeValues: IInvoice

    beforeEach(() => {
      // Here's my submitHandler mock that isn't getting called
      handleSubmit = jest.fn()

      const getComponent = () => (
        <LocalizationProvider dateAdapter={AdapterMoment}>
          <InvoiceForm {...props} />
        </LocalizationProvider>
      )
      const wrapper = render(getComponent())
      container = wrapper.container

      fakeValues = {
        no: '100500',
        file: '',
        seller: {
          name: 'Konyaka',
          country: 'Poland',
          address_line_1: 's1',
          address_line_2: 's2',
          address_line_3: 's3',
        },
        buyer: {
          name: 'skotynyaka',
          country: 'Poland',
          address_line_1: 'b1',
          address_line_2: 'b2',
          address_line_3: 'b3',
        },
        dates: {
          issue: '20/07/2022',
          end: '20/07/2022',
          due: '20/07/2022',
        },
        payment: 'cash',
        items: [defaultItem],
      }
      no = container.querySelector('input[name="no"]') as HTMLInputElement
      sellerName = container.querySelector(
        'input[name="seller.name"]'
      ) as HTMLInputElement
      sellerAddress1 = container.querySelector(
        'input[name="seller.address_line_1"]'
      ) as HTMLInputElement
      sellerAddress2 = container.querySelector(
        'input[name="seller.address_line_2"]'
      ) as HTMLInputElement
      sellerAddress3 = container.querySelector(
        'input[name="seller.address_line_3"]'
      ) as HTMLInputElement
      sellerCountry = container.querySelector(
        'input[name="seller.name"]'
      ) as HTMLInputElement
      buyerName = container.querySelector(
        'input[name="buyer.name"]'
      ) as HTMLInputElement
      buyerAddress1 = container.querySelector(
        'input[name="buyer.address_line_1"]'
      ) as HTMLInputElement
      buyerAddress2 = container.querySelector(
        'input[name="buyer.address_line_2"]'
      ) as HTMLInputElement
      buyerAddress3 = container.querySelector(
        'input[name="buyer.address_line_3"]'
      ) as HTMLInputElement
      buyerCountry = container.querySelector(
        'input[name="buyer.name"]'
      ) as HTMLInputElement
      paymentMethod = container.querySelector(
        'input[name="payment"]'
      ) as HTMLInputElement
      dueDate = container.querySelector('.dates_due input') as HTMLInputElement
      issueDate = container.querySelector(
        '.dates_issue input'
      ) as HTMLInputElement
      endDate = container.querySelector('.dates_end input') as HTMLInputElement
      form = container.querySelector('form') as HTMLFormElement

      // Act--------------
      // Change the input values
      act(() => {
        fireEvent.change(no, { target: { value: fakeValues.no } })
        fireEvent.change(sellerName, {
          target: { value: fakeValues.seller.name },
        })
        fireEvent.change(sellerAddress1, {
          target: { value: fakeValues.seller.address_line_1 },
        })
        fireEvent.change(sellerAddress2, {
          target: { value: fakeValues.seller.address_line_2 },
        })
        fireEvent.change(sellerAddress3, {
          target: { value: fakeValues.seller.address_line_3 },
        })
        fireEvent.change(sellerCountry, {
          target: { value: fakeValues.seller.country },
        })
        fireEvent.change(buyerName, {
          target: { value: fakeValues.buyer.name },
        })
        fireEvent.change(buyerAddress1, {
          target: { value: fakeValues.buyer.address_line_1 },
        })
        fireEvent.change(buyerAddress2, {
          target: { value: fakeValues.buyer.address_line_2 },
        })
        fireEvent.change(buyerAddress3, {
          target: { value: fakeValues.buyer.address_line_3 },
        })
        fireEvent.change(buyerCountry, {
          target: { value: fakeValues.buyer.country },
        })
        fireEvent.change(paymentMethod, {
          target: { value: fakeValues.payment },
        })
        fireEvent.change(issueDate, {
          target: { value: fakeValues.dates.issue },
        })
        fireEvent.change(dueDate, { target: { value: fakeValues.dates.due } })
        fireEvent.change(endDate, { target: { value: fakeValues.dates.end } })

        // This should submit the form?
        fireEvent.submit(form)
      })
    })

    test('Submit', async () => {
      // Assert--------------
      act(async () => {
        await waitFor(() => {
          expect(handleSubmit).toHaveBeenCalledTimes(1)
          expect(handleSubmit).toHaveBeenCalledWith(fakeValues)
        })
      })
    })
  })
})
