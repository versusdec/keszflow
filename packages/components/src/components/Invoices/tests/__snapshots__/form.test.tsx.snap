// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Invoice Form render Should render correctly with Shallow 1`] = `ShallowWrapper {}`;

exports[`Invoice Form render Should render correctly with mount 1`] = `ReactWrapper {}`;
